hazelcast/src/main/template/com/hazelcast/instance/GeneratedBuildProperties.java

```diff
@@ -28,9 +30,9 @@
  * Hazelcast deployed.
  *
  * WARNING: DO NOT CHANGE FIELD NAMES IN THE TEMPLATE.
- * The fields are read via reflection at {@link com.hazelcast.instance.BuildInfoProvider}
  *
  */
+@Generated("com.hazelcast.instance.BuildInfoProvider")
```

Comment:
nit-pick: the `@Generated` contract specifies that 

> The value element must have the name of the code generator. The recommended convention is to use the fully qualified name of the code generator. For example: com.acme.generator.CodeGen.

`BuildInfoProvider` reads the info in this class but its generation occurs via `maven-resources-plugin`, so `BuildInfoProvider` isn't really its generator.

I think I 'd rather leave the annotation value blank and add the reference to `BuidlInfoProvider` class either as it was or with a `@see` javadoc annotation.